domain-tier/src/main/java/com/outdoor/foodcalc/domain/service/dish/DishDomainService.java

```diff
@@ -75,14 +75,15 @@ public Dish addDish(Dish dish) {
         if(id == -1L) {
             throw new FoodcalcDomainException("Failed to add dish");
         }
-        if((dish.getProducts().size() > 0) && (!productRefRepo.addDishProducts(dish))) {
-            throw new FoodcalcDomainException("Fail to add products for dish with id=" + dish.getDishId());
-        }
-        return new Dish(id,
+        Dish addedDish = new Dish(id,
                 dish.getName(),
                 dish.getDescription(),
                 dish.getCategory(),
                 dish.getProducts());
+        if((addedDish.getProducts().size() > 0) && (!productRefRepo.addDishProducts(addedDish))) {
```

Comment:
Let's revert this function back
I mean add dish products and return new dish object after that

Reply:
done